app/components/Account/CollateralTable.jsx

```diff
@@ -0,0 +1,255 @@
+ */
+class ListGenerator extends React.Component {
+    static propTypes = {
+        callOrders: ChainTypes.ChainObjectsList
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            assetsPropsCount: 0,
+            callOrdersCount: 0
+        };
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        // if we got new bitasset info
+        // or call orders changed
+        // then recalculate "cache" of maring items
+
+        const callOrdersJson = JSON.stringify(nextProps.callOrders);
+
+        if (
+            prevState == null ||
+            prevState.assetsPropsCount != nextProps.bitAssets.length ||
+            prevState.ordersJson != callOrdersJson
+        ) {
+            let assets = new Array();
+            // construct map of bitassets
+            let assetsMap = {};
+            // index to track asset info
+            let index = 0;
+
+            nextProps.bitAssets.forEach(o => {
+                assetsMap[o.get("id")] = index++;
+                assets.push({
+                    asset: o,
+                    has_margin_order: false
+                });
+            });
+
+            if (nextProps.callOrders.length > 0) {
+                // add to iterated bitasets items that wasn't listed by default (component's callOrders prop)
+                nextProps.callOrders.forEach(o => {
+                    let assetId = o.getIn(["call_price", "quote", "asset_id"]);
+
+                    if (assetsMap[assetId] == null) {
+                        let newAssetInfo = ChainStore.getObject(assetId);
+                        if (typeof newAssetInfo != "undefined") {
+                            assetsMap[assetId] = index++;
+                            assets.push({
+                                asset: newAssetInfo,
+                                has_margin_order: true,
+                                order: o
+                            });
+                        }
+                    } else {
+                        // mark as margin order
+                        assets[assetsMap[assetId]].has_margin_order = true;
+                        assets[assetsMap[assetId]].order = o;
+                    }
+                });
+            }
+
+            return {
+                assetsPropsCount: nextProps.bitAssets.length,
+                assets: assets,
+                ordersJson: callOrdersJson
+            };
+        }
+
+        return prevState;
+    }
+
+    render() {
+        let {account, callOrders} = this.props;
+        callOrders = callOrders.filter(o => !!o);
+
+        let rows = this.state.assets
+            .sort((a, b) => {
+                // a,b - order ChainObject's
+                // could be done via component's property `order_by`
+
+                let aVal = a.has_margin_order ? a.order.get("debt") : 0;
+                let bVal = b.has_margin_order ? b.order.get("debt") : 0;
+
+                return bVal - aVal;
```

Comment:
if both have no order, fallback to sort by symbol

Reply:
ok

Reply:
hm, then i'll add 0 before symbol in order to keep order of rows with debt, @sschiessl-bcp what do you think?

Reply:
Sorry for being vague.

Sorting:
 - having an order goes above having no order
 - if both have an order, sort by debt
 - if both have no order, sort by symbol